src/app/services/article.service.ts

```diff
@@ -81,6 +78,13 @@ export class ArticleService {
       if (this.articleMap.get(articleId)) {
         return this.articleMap.get(articleId)!;
       }
+      // Fix for not fetching hard coded items until not needed.
```

Comment:
My intent was initially to replace "this.article" (an array) with "this.articleMap" (a map). 

They serve the same purpose. I think this.article needs to be a map to simplify access to the data, i.e. without having to loop through it (which is what `find()` does).

Anyway, I'm not opposed to having both. But I think the solution should be to populate this.articleMap with the hardcoded article with documentId = this.NEW_LABEL (this probably belongs to a AppConstants of sorts class).

Maybe pre-populate it with data from `getHardcodedArticles()`.

hardcoded articles is a stop-gap. Once we have all articles have the necessary elements like the image, we can drop it, so, we can make this change then?

Reply:
The primary purpose of this.articles was not searching. Angular is built on Change detection that works with primitive and references. When the array of articles changes, angular understands to repaint its rendering.

We can have both, one for the O(1) lookup, and one for angular's one-way-data-flow change detection.